File: frontend/src/types/job.ts
export type JobStatus = 'PENDING' | 'RUNNING' | 'COMPLETED' | 'FAILED';
export type JobType = 'TEST_JOB' | 'DATA_INGESTION' | 'DATA_ANALYSIS' | 'MODEL_TRAINING';

export interface Job {
  id: string;
  job_type: JobType;
  status: JobStatus;
  created_at: string;
  started_at?: string | null;
  completed_at?: string | null;
  error_message?: string | null;
  result_path?: string | null;
}

export interface JobListResponse {
  total: number;
  items: Job[];
}

File: backend/app/schemas/job.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from app.db.models import JobStatus, JobType


class JobCreate(BaseModel):
    job_type: JobType = JobType.TEST_JOB


class JobResponse(BaseModel):
    id: str
    job_type: JobType
    status: JobStatus
    created_at: datetime
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    error_message: Optional[str] = None
    result_path: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class JobListResponse(BaseModel):
    total: int
    items: list[JobResponse]
